{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79b84844",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ENIGMA Pre Imputation Code\n",
    "\n",
    "$HOME/enigma/DTIgenetics/\n",
    "mkdir -p $HOME/enigma/DTIgenetics/\n",
    "\n",
    "cd $HOME/enigma/DTIgenetics/\n",
    "\n",
    "# SNP filtering and Pruning for accurate sex check\n",
    "\n",
    "plink --bfile ${datafile} --mind 1 --hwe 1e-06 --geno 0.01 --maf 0.05 --make-bed --out ${datafile}_filtered\n",
    "plink --bfile ${datafile}_filtered --indep-pairphase 20000 2000 0.5 --out ${datafile}_pruned\n",
    "plink --bfile ${datafile}_filtered --extract ${datafile}_pruned.prune.in --make-bed --out ${datafile}_pruned\n",
    "\n",
    "# Split-X to deal with pseudo-autosomal regions\n",
    "awk '{if ($1==25) print $1}' ${datafile}_pruned.bim | wc -l\n",
    "\n",
    "plink --bfile ${datafile}_pruned --split-x b37 no-fail --make-bed --out ${datafile}_splitX\n",
    "\n",
    "plink --bfile ${datafile}_splitX --check-sex 0.2 0.8 --out ${datafile}_sexcheck\n",
    "\n",
    "grep PROBLEM ${datafile}_sexcheck.sexcheck > sex.drop\n",
    "\n",
    "wc -l sex.drop > ${datafile}_sexcheck_PROBLEM.txt\n",
    "\n",
    "plink --bfile ${datafile} --remove sex.drop --make-bed --out ${datafile}_QC1\n",
    "\n",
    "awk '{if ($1==\"23\") print $0}' ${datafile}_QC1.bim | wc -l > snp_count_X.txt\n",
    "echo \"The number of SNPs on the X chromosome is:\"\n",
    "cat snp_count_X.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "101ac9cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check for related and duplicated samples\n",
    "plink --bfile ${datafile}_QC1 --mind 1 --geno 0.01 --maf 0.05 --make-bed --out ${datafile}_QC1tmp\n",
    "plink --bfile ${datafile}_QC1tmp --indep-pairwise 100 5 0.2 --out ${datafile}_QC1pruned\n",
    "plink --bfile ${datafile}_QC1tmp --extract ${datafile}_QC1pruned.prune.in --make-bed --out ${datafile}_QC1pruned\n",
    "\n",
    "plink --bfile ${datafile}_QC1 --remove pihat_duplicates.txt --make-bed --out ${datafile}_QC2\n",
    "\n",
    "wc -l pihat_duplicates.txt > ${datafile}_QC1pruned_duplicates_count.txt\n",
    "\n",
    "wc -l pihat_relatedness.genome > ${datafile}_QC1pruned_relatedness_count.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "165b0550",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Multi-Dimensional Scaling (MDS) Protocol\n",
    "wget https://enigma.ini.usc.edu/website_downloads/ENIGMA_DTI_downloads/HapMap3/HM3_b37.bed.gz\n",
    "wget https://enigma.ini.usc.edu/website_downloads/ENIGMA_DTI_downloads/HapMap3/HM3_b37.bim.gz\n",
    "wget https://enigma.ini.usc.edu/website_downloads/ENIGMA_DTI_downloads/HapMap3/HM3_b37.fam.gz\n",
    "\n",
    "plink --bfile ${datafile}_QC2 --mind 1 --hwe 1e-6 --geno 0.05 --maf 0.01 --make-bed --out ${datafile}_QC2_filtered\n",
    "\n",
    "gunzip HM3_b37.*\n",
    "awk '{print $2}' HM3_b37.bim > HM3_b37.snplist.txt\n",
    "plink --bfile ${datafile}_QC2_filtered --extract HM3_b37.snplist.txt --make-bed --out ${datafile}_QC2local\n",
    "awk '{ if (($5==\"T\" && $6==\"A\")||($5==\"A\" && $6==\"T\")||($5==\"C\" && $6==\"G\")||($5==\"G\" && $6==\"C\")) print $2, \"ambig\"; else print $2 ;}' ${datafile}_QC2_filtered.bim | grep -v ambig > local.snplist.txt\n",
    "plink --bfile HM3_b37 --extract local.snplist.txt --make-bed --out HM3_b37_external\n",
    "\n",
    "plink --bfile ${datafile}_QC2local --bmerge HM3_b37_external.bed HM3_b37_external.bim HM3_b37_external.fam --make-bed --out ${datafile}_QC2local_HM3b37merge\n",
    "\n",
    "plink --bfile ${datafile}_QC2local --flip ${datafile}_QC2local_HM3b37merge-merge.missnp --make-bed --out ${datafile}_QC2local_flipped\n",
    "plink --bfile ${datafile}_QC2local_flipped --bmerge HM3_b37_external.bed HM3_b37_external.bim HM3_b37_external.fam --make-bed --out ${datafile}_QC2local_HM3b37merge\n",
    "\n",
    "plink --bfile ${datafile}_QC2local_flipped --exclude  ${datafile}_QC2local_HM3b37merge-merge.missnp --make-bed --out ${datafile}_QC2local_flipped2\n",
    "plink --bfile ${datafile}_QC2local_flipped2 --bmerge HM3_b37_external.bed HM3_b37_external.bim HM3_b37_external.fam --make-bed --out ${datafile}_QC2local_HM3b37merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfa347ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run the MDS analysis\n",
    "plink --bfile ${datafile}_QC2local_HM3b37merge --cluster --mind .05 --mds-plot 10 --extract local.snplist.txt --out ${datafile}_QC2_HM3b37mds\n",
    "\n",
    "# Plot the MDS results using R\n",
    "cat ${datafile}_QC2_HM3b37mds.mds | tr -s ' ' '\\t' > ${datafile}_QC2_HM3b37.mds.tsv\n",
    "awk 'BEGIN{OFS=\",\"};{print $1, $2, $3, $4, $5, $6, $7}' >> ${datafile}_QC2_HM3b37mds2R.mds.csv ${datafile}_QC2_HM3b37mds.mds\n",
    "\n",
    "module load R \n",
    "\n",
    "R \n",
    "\n",
    "#In R\n",
    "library(calibrate)\n",
    "#If you donâ€™t have the calibrate package, install it using\n",
    "#install.packages(\"calibrate\")\n",
    "mds.cluster = read.csv(\"DATAFILE_QC2_HM3b37mds2R.mds.csv\", header=T);\n",
    "#first we create an additional column holding the respective ancestry labels\n",
    "mds.cluster$POP=rep(\"DATASETNAME\",length(mds.cluster$C1));\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"CEU\")] <- \"CEU\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"CHB\")] <- \"CHB\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"YRI\")] <- \"YRI\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"TSI\")] <- \"TSI\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"JPT\")] <- \"JPT\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"CHD\")] <- \"CHD\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"MEX\")] <- \"MEX\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"GIH\")] <- \"GIH\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"ASW\")] <- \"ASW\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"LWK\")] <- \"LWK\";\n",
    "mds.cluster$POP[which(mds.cluster$FID == \"MKK\")] <- \"MKK\";\n",
    "\n",
    "#get the colors for plotting\n",
    "colors=rep(\"red\",length(mds.cluster$C1));\n",
    "colors[which(mds.cluster$POP == \"CEU\")] <- \"lightblue\";\n",
    "colors[which(mds.cluster$POP == \"CHB\")] <- \"brown\";\n",
    "colors[which(mds.cluster$POP == \"YRI\")] <- \"yellow\";\n",
    "colors[which(mds.cluster$POP == \"TSI\")] <- \"green\";\n",
    "colors[which(mds.cluster$POP == \"JPT\")] <- \"purple\";\n",
    "colors[which(mds.cluster$POP == \"CHD\")] <- \"orange\";\n",
    "colors[which(mds.cluster$POP == \"MEX\")] <- \"grey50\";\n",
    "colors[which(mds.cluster$POP == \"GIH\")] <- \"black\";\n",
    "colors[which(mds.cluster$POP == \"ASW\")] <- \"darkolivegreen\";\n",
    "colors[which(mds.cluster$POP == \"LWK\")] <- \"magenta\";\n",
    "colors[which(mds.cluster$POP == \"MKK\")] <- \"darkblue\";\n",
    "\n",
    "pdf(file=\"mdsplot_DATASETNAME_QC2_outliersincluded.pdf\",width=7,height=7)\n",
    "\n",
    "#Please note - the plot will directly be saved as a PDF in your current working directory under the above specified name\n",
    "#and does not appear in the PLOT window\n",
    "plot(rev(mds.cluster$C2), rev(mds.cluster$C1), col=rev(colors), ylab=\"Dimension 1\", xlab=\"Dimension 2\",pch=20)\n",
    "minor.tick(nx = 5, ny = 5,   # Ticks density\n",
    "         tick.ratio = 0.5) # Ticks size\n",
    "legend(\"bottomleft\", c(\"DATASETNAME\", \"CEU\", \"CHB\", \"YRI\", \"TSI\", \"JPT\", \"CHD\", \"MEX\", \"GIH\", \"ASW\",\"LWK\", \"MKK\"), fill=c(\"red\", \"lightblue\", \"brown\", \"yellow\", \"green\", \"purple\", \"orange\", \"grey50\", \"black\", \"darkolivegreen\", \"magenta\", \"darkblue\"))\n",
    "dev.off();\n",
    "#If you want to know the subject ID label of each sample on the graph, you can label your sample points by uncommenting the commands below.\n",
    "#This is optional and you can choose not to do this if you are worried about patient information being sent; when you send us your MDS plot please make sure the subject ID labels are NOT on the graph.\n",
    "#POPlabels <- c(\"CEU\", \"CHB\", \"YRI\", \"TSI\", \"JPT\", \"CHD\", \"MEX\", \"GIH\", \"ASW\", \"LWK\", \"MKK\");\n",
    "#textxy(mds.cluster[which(!(mds.cluster$POP %in% POPlabels)), \"C2\"], mds.cluster[which(!(mds.cluster$POP %in% POPlabels)), \"C1\"], mds.cluster[which(!(mds.cluster$POP %in% POPlabels)), \"IID\"])\n",
    "\n",
    "# Identify Ancestry Outliers\n",
    "\n",
    "MDS_mySample <- mds.cluster[which(mds.cluster$POP %in% c(\"`DATASETNAME\")),]\n",
    "MDS_mySample$outlier  <- 0\n",
    "# calculate outliers - first a threshold example for the g1000mix, a quite homogenous sample\n",
    "# NOTE: change the cut-off below depending on YOUR sample \n",
    "MDS_mySample$outlier[MDS_mySample$C1 < -0.01 | MDS_mySample$C2 > -0.05] <- 1\n",
    "\n",
    "MDS_mySample_outliers<- MDS_mySample[which(MDS_mySample$outlier == 1),1:2]\n",
    "MDS_mySample_european<- MDS_mySample[which(MDS_mySample$outlier == 0 ),1:2]\n",
    "#For further plotting we also grab the full reference details again and combine them with the full EU information in your sample\n",
    "#First let's get the reference - we get everything not part of your dataset\n",
    "MDS_ref_eur <- mds.cluster[which \"(!(mds.cluster$POP %in% \"DATASETNAME\")),]\n",
    "#combine this with all Europeans (or respective other ancestry cohort)\n",
    "MDS_mySample_eur_plus_ref <- rbind(MDS_ref_eur, mds.cluster[which (mds.cluster$IID %in% MDS_mySample_european$IID ),])\n",
    "# repeat the colours again for the updated dataframe and then plot - note again: plot is directly saved\n",
    "colors=rep(\"red\",length(MDS_mySample_eur_plus_ref$C1));\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"CEU\")] <- \"lightblue\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"CHB\")] <- \"brown\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"YRI\")] <- \"yellow\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"TSI\")] <- \"green\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"JPT\")] <- \"purple\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"CHD\")] <- \"orange\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"MEX\")] <- \"grey50\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"GIH\")] <- \"black\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"ASW\")] <- \"darkolivegreen\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"LWK\")] <- \"magenta\";\n",
    "colors[which(MDS_mySample_eur_plus_ref$POP == \"MKK\")] <- \"darkblue\";\n",
    "\n",
    "pdf(file=\"mdsplot_DATASETNAME_QC2_outliersExcluded.pdf\", width=7, height=7)\n",
    "plot(rev(MDS_mySample_eur_plus_ref$C2), rev(MDS_mySample_eur_plus_ref$C1), col=rev(colors), ylab=\"Dimension 1\", xlab=\"Dimension 2\", pch=20)\n",
    "legend(\"bottomleft\", c(\"DATASETNAME\", \"CEU\", \"CHB\", \"YRI\", \"TSI\", \"JPT\", \"CHD\", \"MEX\", \"GIH\", \"ASW\",\"LWK\", \"MKK\"), fill=c(\"red\", \"lightblue\", \"brown\", \"yellow\", \"green\", \"purple\", \"orange\", \"grey50\", \"black\", \"darkolivegreen\", \"magenta\",` `\"darkblue\"))\n",
    "# save files - don't forget to replace DATASETNAME with your cohort-specific abbreviation\n",
    "#let's get the outliers in your sample\n",
    "write.table(MDS_mySample_outliers, \"DATASETNAME_pop_strat_mds.outlier.txt\", sep=\"\\t\", quote=FALSE, row.names=FALSE)\n",
    "#let's get the europeans in your sample\n",
    "write.table(MDS_mySample_european, \"DATASETNAME_pop_strat_mds.eur.txt\", sep=\"\\t\", quote=FALSE, row.names=FALSE)\n",
    "# quit R\n",
    "q()\n",
    "\n",
    "plink --bfile ${datafile}_QC2_filtered  --keep ${datafile}_pop_strat_mds.eur.txt --make-bed --out ${datafile}_QC3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e20ec91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get genetic principal components in ancestry homogenous group\n",
    "plink --bfile ${datafile}_QC3 --pca --extract local.snplist.txt --out ${datafile}_PCACovariates\n",
    "\n",
    "R\n",
    "\n",
    "library(data.table)\n",
    "library(ggplot2)\n",
    "\n",
    "#read in the eigenvalue file\n",
    "eigv <- fread(\"${datafile}_PCACovariates.eigenval\")\n",
    "#calculate variance explained\n",
    "var_explained <- (eigv/(sum(eigv))*100\n",
    "#plot the scree plot\n",
    "qplot(c(1:20), var_explained$V1) +\n",
    "geom_line() +\n",
    "xlab(\"Principal Component\") +\n",
    "ylab(\"Variance Explained\") +\n",
    "ggtitle(\"Scree Plot\") +\n",
    "ylim(0, 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "91958912",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Cohort Summary Data \n",
    "\n",
    "# Pre-QC subject numbers\n",
    "echo \"COHORTNAME N_CONTROLS N_CASES N_CONTROLS_M N_CONTROLS_F N_CASES_M N_CASES_F PROP_CONTROLS_F PROP_CONTROLS_M PROP_CASES_F PROP_CASES_M\" > ${datafile}_basic_stats_preQC.txt\n",
    "echo -n \"${datafile} \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#How many controls/unaffected are in the cohort?\n",
    "N_CONTROLS=`awk '{if ($6==1) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#How many patients are in the cohort?\n",
    "N_CASES=`awk '{if ($6==2) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CASES \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#What is the absolute number of males among controls/unaffected?\n",
    "N_CONTROLS_M=`awk '{if ($6==1 && $5==1) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS_M \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#What is the absolute number of females among controls/unaffected?\n",
    "N_CONTROLS_F=`awk '{if ($6==1 && $5==2) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS_F \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#What is the absolute number of males among patients?\n",
    "N_CASES_M=`awk '{if ($6==2 && $5==1) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CASES_M \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#What is the absolute number of females among patients?\n",
    "N_CASES_F=`awk '{if ($6==2 && $5==2) print $0}' ${datafile}.fam | wc -l`\n",
    "echo -n \"$N_CASES_F \" >> ${datafile}_basic_stats_preQC.txt\n",
    "#What is the proportion of females among controls/unaffected?\n",
    "PROP_CONTROLS_F=`echo \"$N_CONTROLS_F/$N_CONTROLS\" | bc -l`\n",
    "#What is the proportion of males among controls/unaffected?\n",
    "PROP_CONTROLS_M=`echo \"$N_CONTROLS_M/$N_CONTROLS\" | bc -l`\n",
    "#What is the proportion of females among patients?\n",
    "PROP_CASES_F=`if (($N_CASES == 0)); then echo 0; else echo \"$N_CASES_F/($N_CASES_F+$N_CASES_M)\" | bc -l; fi`\n",
    "#What is the proportion of males among patients?\n",
    "PROP_CASES_M=`if (($N_CASES == 0)); then echo 0; else echo \"$N_CASES_M/($N_CASES_F+$N_CASES_M)\" | bc -l; fi`\n",
    "echo -n \"$PROP_CONTROLS_F $PROP_CONTROLS_M $PROP_CASES_F $PROP_CASES_M \" >> ${datafile}_basic_stats_preQC.txt\n",
    "\n",
    "\n",
    "# SNP level QC\n",
    "\n",
    "echo \"COHORTNAME N_SNPs_preQC N_samples_preQC SNPs_removed_>missingnessT Samples_removed_>missingnessT SNPs_removed_<MAFt SNPs_removed_HWE Samples_removed_MDS_outliers N_SNPs_postQC N_samples_postQC\" > ${datafile}_qc_summary.txt\n",
    "echo -n \"${datafile} \" >> ${datafile}_qc_summary.txt\n",
    "NLINES=`wc -l ${datafile}.bim | awk '{print $1}'`\n",
    "echo -n $NLINES\" \" >> ${datafile}_qc_summary.txt\n",
    "NLINES=`wc -l ${datafile}.fam | awk '{print $1}'`\n",
    "echo -n $NLINES\" \" >> ${datafile}_qc_summary.txt\n",
    "qc_VAR=$(grep \"variants removed due to missing genotype data (--geno)\" ${datafile}_QC2_filtered.log | awk '{printf $1}')\n",
    "if [ -z \"$qc_VAR\" ];then echo -n \"NaN\"\" \" >> ${datafile}_qc_summary.txt; else echo -n \"$qc_VAR\"\" \" >> ${datafile}_qc_summary.txt;fi\n",
    "qc_VAR=$(grep \"people removed due to missing genotype data (--mind)\" ${datafile}_QC2_filtered.log | awk '{printf $1}')\n",
    "if [ -z \"$qc_VAR\" ];then echo -n \"NaN\"\" \" >> ${datafile}_qc_summary.txt; else echo -n \"$qc_VAR\"\" \" >> ${datafile}_qc_summary.txt;fi\n",
    "qc_VAR=$(grep \"variants removed due to minor allele threshold(s)\" ${datafile}_QC2_filtered.log | awk '{printf $1}')\n",
    "if [ -z \"$qc_VAR\" ];then echo -n \"NaN\"\" \" >> ${datafile}_qc_summary.txt; else echo -n \"$qc_VAR\"\" \" >> ${datafile}_qc_summary.txt;fi\n",
    "qc_VAR=$(grep \"variants removed due to Hardy-Weinberg exact test\" ${datafile}_QC2_filtered.log | awk '{printf $2}')\n",
    "if [ -z \"$qc_VAR\" ];then echo -n \"NaN\"\" \" >> ${datafile}_qc_summary.txt; else echo -n \"$qc_VAR\"\" \" >> ${datafile}_qc_summary.txt;fi\n",
    "NLINES2=`wc -l ${datafile}_QC2.bim | awk '{print $1}'`\n",
    "NLINES3=`wc -l ${datafile}_QC3.bim | awk '{print $1}'`\n",
    "N_excluded_MDS_outliers=`echo \"$NLINES2-$NLINES3\" | bc -l`\n",
    "echo -n $N_excluded_MDS_outliers\" \" >> ${datafile}_qc_summary.txt\n",
    "NLINES=`wc -l ${datafile}_QC3.bim | awk '{print $1}'`\n",
    "echo -n $NLINES\" \" >> ${datafile}_qc_summary.txt\n",
    "NLINES=`wc -l ${datafile}_QC3.fam | awk '{print $1}'`\n",
    "echo -n $NLINES\" \" >> ${datafile}_qc_summary.txt\n",
    "\n",
    "# Post-QC subject numbers\n",
    "\n",
    "echo \"COHORTNAME N_CONTROLS N_CASES N_CONTROLS_M N_CONTROLS_F N_CASES_M N_CASES_F PROP_CONTROLS_F PROP_CONTROLS_M PROP_CASES_F PROP_CASES_M\" > ${datafile}_basic_stats_postQC.txt\n",
    "echo -n \"${datafile} \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#How many controls/unaffected are in the cohort?\n",
    "N_CONTROLS=`awk '{if ($6==1) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#How many patients are in the cohort?\n",
    "N_CASES=`awk '{if ($6==2) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CASES \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#What is the absolute number of males among controls/unaffected?\n",
    "N_CONTROLS_M=`awk '{if ($6==1 && $5==1) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS_M \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#What is the absolute number of females among controls/unaffected?\n",
    "N_CONTROLS_F=`awk '{if ($6==1 && $5==2) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CONTROLS_F \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#What is the absolute number of males among patients?\n",
    "N_CASES_M=`awk '{if ($6==2 && $5==1) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CASES_M \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#What is the absolute number of females among patients?\n",
    "N_CASES_F=`awk '{if ($6==2 && $5==2) print $0}' ${datafile}_QC3.fam | wc -l`\n",
    "echo -n \"$N_CASES_F \" >> ${datafile}_basic_stats_postQC.txt\n",
    "#What is the proportion of females among controls/unaffected?\n",
    "PROP_CONTROLS_F=`echo \"$N_CONTROLS_F/$N_CONTROLS\" | bc -l`\n",
    "#What is the proportion of males among controls/unaffected?\n",
    "PROP_CONTROLS_M=`echo \"$N_CONTROLS_M/$N_CONTROLS\" | bc -l`\n",
    "#What is the proportion of females among patients?\n",
    "PROP_CASES_F=`if (($N_CASES == 0)); then echo 0; else echo \"$N_CASES_F/($N_CASES_F+$N_CASES_M)\" | bc -l; fi`\n",
    "#What is the proportion of males among patients?\n",
    "PROP_CASES_M=`if (($N_CASES == 0)); then echo 0; else echo \"$N_CASES_M/($N_CASES_F+$N_CASES_M)\" |  bc -l; fi`\n",
    "echo -n \"$PROP_CONTROLS_F $PROP_CONTROLS_M $PROP_CASES_F $PROP_CASES_M \" >> ${datafile}_basic_stats_postQC.txt"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "name": "python"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
